core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1319,4 +1350,34 @@ private boolean rerunWF(String workflowId, String taskId, Map<String, Object> ta
         }
         return false;
     }
+
+    private boolean acquireLock(String workflowId) {
+        if (config.enableWorkflowExecutionLock()) {
+            if (!lockService.acquireLock(workflowId, 2, TimeUnit.MILLISECONDS)) {
+                LOGGER.info("Thread {} failed to acquire lock to workflowId {}.", Thread.currentThread().getId(), workflowId);
+                Monitors.recordAcquireLockUnsuccessful(workflowId);
+                return false;
+            }
+            LOGGER.info("Thread {} acquired lock to workflowId {}.", Thread.currentThread().getId(), workflowId);
+        }
+        return true;
+    }
+
+    /**
+     * Blocks until it gets the lock for workflowId
+     * @param workflowId
+     */
+    private void waitForLock(String workflowId) {
+        if (config.enableWorkflowExecutionLock()) {
+            lockService.acquireLock(workflowId);
+            LOGGER.info("Thread {} acquired lock to workflowId {}.", Thread.currentThread().getId(), workflowId);
+        }
+    }
+
+    private void releaseLock(String workflowid) {
```

Comment:
Consider sticking to camelCase for workflowId in the param

Reply:
typo... good catch.